apps/rush-lib/src/api/RushConfigurationProject.ts

```diff
@@ -123,7 +123,7 @@ export class RushConfigurationProject {
       );
     }
 
-    this._packageJsonEditor = PackageJsonEditor.load(packageJsonFilename);
+    this._packageJsonEditor = PackageJsonEditor.fromObject(this._packageJson, packageJsonFilename);
```

Comment:
@octogonz Are you familiar with this part of code? Do you think it is safe to reuse this object here?

Reply:
`PackageJsonEditor` was a system that @nickpape had started working on, to provide a model for how config files can be loaded, modified, and then resaved only if changes were made, and normalized during saving.

Prior to your change the `RushConfiguration` state was separate from the `packageJsonEditor` state -- you are combining them to be the same object.  I think that is probably safe?  You would need to test the Rush operations that reference this object, which are basically `rush publish`, `rush version`, `rush update` I think.

I feel like your change here is probably safe.

Reply:
I spent some time reading the code. It looks like a safe change.

Reply:
Also, in RushConfigurationProject, the packageJson property is marked as deprecated with the suggestion "use packageJsonEditor instead".